build/rust/rust_cxx.gni

```diff
@@ -0,0 +1,101 @@
+# Copyright 2021 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//brave/build/cargo.gni")
+
+set_defaults("rust_cxx") {
+  native_header_deps = []
+}
+
+# Template to build Rust/C++ bindings using the tooling
+# described at https://cxx.rs.
+#
+# (Specifically, this builds the C++ side of the bindings.
+# The Rust side of the bindings are not 'built' as such, but
+# automatically expanded by the Rust compiler by virtue of a
+# proc macro which is a simple dependency relationship on the
+# Rust target).
+#
+# This template expands to a source_set containing the
+# C++ side of the bindings. Simply treat it as a dependency.
+#
+# If you're using this, consider whether you should be using
+# rust_source_set.gni or mixed_source_set.gni which have
+# facilities to do the same code generation in an easier
+# way.
+#
+# Parameters:
+#
+# inputs:
+#   The .rs files containing a #[cxx::bridge] section.
+#
+# native_header_deps (optional)
+#   C++ targets on which the headers depend in order to build
+#   successfully.
+#
+# Consumers of this template may alternatively depend upon
+# the :${target_name}_gen target if you wish to include
+# the generated source code in an *existing* source set.
+# To do this:
+# * depend on the target name with a _gen suffix;
+# * for each of the inputs, include in your sources:
+#     file_name.rs.h
+#     file_name.rs.cc
+# * ensure you also depend upon //build/rust:cxx_cppdeps
+#
+# A good way to get the file names is:
+#   process_file_template(list-of-bindings-rs-files,
+#       [ "{{source_gen_dir}}/{{source_file_part}}.h",
+#         "{{source_gen_dir}}/{{source_file_part}}.cc" ]
+#
+# You might want to depend on this template in this mode if
+# you have an existing source_set which is referred to
+# using 'visibility' rules throughout the codebase.
+#
+template("rust_cxx") {
+  assert(defined(invoker.input), "Must specify the Rust file to use as input.")
+  action_foreach("${target_name}_gen") {
+    cxxbridge_executable = "//brave/build/rustup/0.1.8/bin/cxxbridge"
```

Comment:
I don't think this versioned path should be hardcoded here - it seems likely to get out of date when we update our Rust toolchain

Reply:
you are correct and I thought I had a TODO here

Reply:
I think we don't need to pass this at all anymore now that installed from cargo and we can get the path in python

Reply:
marking as resolved; this seems to have been removed